src/common/uilocale.cpp

```diff
@@ -520,7 +520,14 @@ const wxUILocale& wxUILocale::GetCurrent()
     // We initialize it on demand.
     if ( !ms_current.m_impl )
     {
-        ms_current = wxUILocale(wxUILocaleImpl::CreateStdC());
+        // Instantiate standard C locale (should never fail) ...
```

Comment:
I'm not 100% sure that calling `Use()` on C locale is a good idea. At least under Unix it's not a NOP.

Why not set `g_nsloc` in Mac-specific `CreateStdC()` instead?

Reply:
> I'm not 100% sure that calling `Use()` on C locale is a good idea. At least under Unix it's not a NOP.

Yes, `wxUILocale::Use()` is not a NOP under Unix. It sets the C runtime locale via `setlocale`. **But** does that matter in our case? IMHO **no**, because we would call `wxUILocale::Use()` only, if `wxUILocale::Use()` was **not** called. In that case we can assume that the current C runtime locale is the "**C**" locale anyway. Setting it in `wxUILocale::Use()` wouldn't cause any harm, would it?

> Why not set `g_nsloc` in Mac-specific `CreateStdC()` instead?

Unfortunately, this is not possible. `CreateStdC()` can be called from various places, not only from `Use()` or `GetCurrent()`, but also from the `wxUILocale()` constructor. Only in `Use()` you can be sure that the locale is the **current** locale.

Reply:
You're probably right, but the trouble is that it's not immediately obvious and for me it's a bad sign when you need to carefully think about something instead of it just being immediately clear.

And it just seems suspicious to call `Use()` from an accessor. In fact I am still not sure to understand why do we need to remember this locale as the default one here, i.e. why does querying the default locale should change anything?

Reply:
> You're probably right, but the trouble is that it's not immediately obvious and for me it's a bad sign when you need to carefully think about something instead of it just being immediately clear.

I fully agree. And you asked for a more detailed explanation in the general comments. However, I hope I could make my point in my answer there.

> And it just seems suspicious to call `Use()` from an accessor. In fact I am still not sure to understand why do we need to remember this locale as the default one here, i.e. why does querying the default locale should change anything?

Well, actually it is the same as with any method that instantiates a static global object when called for the first time: it checks whether the global object already exists and is initialized; if yes, a reference to the global object is returned; if no, the global object will allocated and initialized, before a reference to it will be returned.

Here we have the situation that on the **first** call to `GetCurrent()` the global object `ms_current` is not initialized, because `UseDefault()` was not called. One option would have been to call `UseDefault()` from `GetCurrent()`, because it was not done before by the application. However, that would force the application to use the default user locale, but it is unlikely that that is the intention of the application, when it did not call `UseDefault()`. More likely is that the application wants to stay with the standard C locale. Therefore we would need a method `UseStdC()` to make clear that we want to _**use** the standard C locale_. But we don't have it. Therefore `GetCurrent()` calls `CreateStdC()` ... and misses to call `Use()` for it (which would be part of `UseStdC()`. That is adding the call to `Use()` actually fixes a bug.